### PR TITLE
Componentize node view

### PR DESCRIPTION
**Pull request by apostolidhs**
* Replaced relative paths in dashboard build.
* Transform the charts to the menu structure.
* The menu structure contains all the required sections (head charts, title, icons etc..), sorted ready to be displayed by the components.
* Created React contexts in order to store the menu structure.
* Created the sidebar using menu context and composition.

## Notes
* The alias should be in a different PR, sorry for this.
* In order to make the sidebar work properly in the current ecosystem there are some minor compromises like the href ids etc. It Will be fully replaced once we refactor the main node-view screen
* Take a look it is not completely finished yet. I want to exhausted test it on cloud's single node & room overview screen.

### CODE REVIEW

File: src/domains/charts/providers/menu/menu.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React from "react"
+import styled from "styled-components"
+import { Flex } from "@netdata/netdata-ui"
+import { MenuGroupContainer } from "domains/charts/providers/menuGroup"
+import { ActiveMenuConsumer } from "domains/charts/providers/active"
+import { MenuConsumer } from "./context"
+
+export const Menu = ({ ids, activeMenuId, onMenuItemClick, onGroupItemClick, ...rest }) => (
+  <Flex as="ul" column {...rest}>
```
Comment (novykh) [inline src/domains/charts/providers/menu/menu.js:9]: if its a `ul` then children should be `li`

File: src/domains/charts/providers/menu/menu.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        />
+      )
+    })}
+  </Flex>
+)
+
+export const MenuSidebar = styled(Flex).attrs({
+  overflow: { vertical: "auto" },
+})`
+  position: sticky;
```
Comment (novykh) [inline src/domains/charts/providers/menu/menu.js:28]: we need to fix those styles. With a flex, flex: 1 and height of 100% on its parent it should work as needed.

File: src/domains/charts/providers/menuGroup/menuGroup.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import styled, { css } from "styled-components"
+import { Text, Flex, getColor } from "@netdata/netdata-ui"
+import { MenuItemContainer } from "domains/charts/providers/subMenu"
+import { MenuGroupConsumer } from "./context"
+
+const MenuGroupLabel = styled(Text).attrs({ color: "border" })`
+  font-weight: "500";
```
Comment (novykh) [inline src/domains/charts/providers/menuGroup/menuGroup.js:8]: Why don't we use the prop strong in here?

**Review by novykh — APPROVED** [review]

File: src/domains/charts/providers/menuGroup/menuGroup.js (hunk 3 above)
Comment (apostolidhs) [reply src/domains/charts/providers/menuGroup/menuGroup.js:8]: I copied the styling of the sidebar as it is, it has font-weight 500 for this list item...

File: src/domains/charts/providers/menu/menu.js (hunk 2 above)
Comment (apostolidhs) [reply src/domains/charts/providers/menu/menu.js:28]: This style and some others will be fixed once i refactor the charts view. Let's leave it as it was for now.

File: src/domains/charts/providers/menu/menu.js (hunk 1 above)
Comment (apostolidhs) [reply src/domains/charts/providers/menu/menu.js:9]: Yes, it is :)

File: src/domains/charts/getChartMenu.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  })
+
+  const emitMultipartMenu = partName => {
+    const menuPattern = parts.length > 2 && parts[1] === partName ? `${part1}_${part2}` : part1
+    return emit({ menuPattern })
+  }
+
+  if (clusterId) return emit({ menu: `Kubernetes ${clusterId}` })
+
+  switch (part1) {
```
[the remark this answers is not in the document]
Comment (apostolidhs) [reply src/domains/charts/getChartMenu.js:34]: We have already introduced many changes here. I suggest leaving it as it is for now until we fully migrate all the `node-views` references to that approach.

File: src/domains/charts/getChartHeads.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      (acc, chartId) => [
+        ...acc,
+        ...generateHeadCharts(type, getChart(chartId), 0).map((head, index) => ({
+          ...parseChartString(head),
+          id: `${type}|${chartId}|${index}`,
+        })),
+      ],
+      []
+    )
+    .forEach(chart => {
```
[the remark this answers is not in the document]
Comment (apostolidhs) [reply src/domains/charts/getChartHeads.js:18]: Merged in one loop.

**Pull request closed without merging**